solver/src/interactions/dummy_web3.rs

```diff
@@ -24,3 +25,19 @@ pub fn dummy_web3() -> Web3<DummyTransport> {
 pub fn dummy_weth(addr: impl Into<H160>) -> WETH9 {
     WETH9::at(&dummy_web3(), addr.into())
 }
+
+pub fn infura(network: impl AsRef<str>) -> shared::Web3 {
+    let infura_project_id =
+        std::env::var("INFURA_PROJECT_ID").expect("Missing INFURA_PROJECT_ID env variable");
+    from_node_url(&format!(
+        "https://{}.infura.io/v3/{}",
+        network.as_ref(),
+        infura_project_id
+    ))
+}
+
+fn from_node_url(node_url: &str) -> shared::Web3 {
+    Web3::new(LoggingTransport::new(
+        Http::new(node_url).expect("transport creation failed"),
+    ))
+}
```

Comment:
Is this stuff really being used for dummy web3 related functionality?

Reply:
Its used for unit tests. I want to move these to a "testing utilities" file but that would touch too much code for this PR and planed on doing it in a follow up.

Reply:
https://github.com/gnosis/gp-v2-services/pull/537